python/python/lance/__init__.py

```diff
@@ -89,7 +89,13 @@ def dataset(
         Approximately, ``n = Total Rows / number of IVF partitions``.
         ``pq = number of PQ sub-vectors``.
     """
-    ds = LanceDataset(uri, version, block_size, commit_lock=commit_lock)
+    ds = LanceDataset(
```

Comment:
Does the index cache have a TTL (I'm looking at the comment above which, to be fair, looks like it isn't part of this PR)?  I don't think it does.